### PR TITLE
Exclude files and folders from linting (closes #35)

### LINKED ISSUE #35: Info: How to include all files but exclude just one specific file or folder?
See title.

### PR DESCRIPTION
**Pull request by Azbagheri**
### Pull Request Checklist
- [ ] I have created an issue prior to creating this pull request
- [ ] I have provided a detailed description and motivation regarding the change below
- [ ] I have updated the test suite and documentation to support my change

### Corresponding Issue

```
<ADD-ISSUE-HERE>
```

### Description of Change

```
<ADD-DESCRIPTION-HERE>
```

### Motivation and Context

```
<ADD-MOTIVATION-HERE>
```

### Testing Steps

```
<ADD-STEPS-HERE>
```

### Risks

```
<ADD-RISKS-HERE>
```

### Additional Information

```
<ADD-ADDITIONAL-INFORMATION-HERE>
```

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -27,48 +27,57 @@ jobs:
...
           path: "src/*.sh"
 ```
+#### Exclude files and folders from the static analysis:
+```yml
+      - name: Run Shellcheck
+        uses: azohra/shell-linter@latest
+        with:
+          excluded_paths: "src/setup.sh,tests/unit_tests" 
+```
+Note that `excluded_paths` only accepts paths relative to your project's root directory. However, **do not** include `./` at the beginning of the paths.
```
Comment (asadmansr) [inline README.md:68]: Really excited for the excluded paths feature. A little decision making on our end regarding the format of the name of this input. Should we go with `excluded_paths` or `excluded-paths`? I saw some examples of input names on the docs ([here](https://docs.github.com/en/actions/creating-actions/creating-a-composite-action) and [here](https://docs.github.com/en/actions/creating-actions/metadata-syntax-for-github-actions))

File: README.md [hunk 2]
```diff
@@ -77,11 +86,12 @@ Run analysis by using a specific version of Shell Linter:
 # Input
 
 ### `path`
-
 Optional. Execute lint check on a specific file or folder. Default: `.`
 
-### `severity`
+### `excluded_paths`
+Optional. Exclude files and folders from Shellcheck scan. Default: scan all the Shellcheck-supported shell scripts (sh/bash/dash/ksh).
```
Comment (asadmansr) [inline README.md:92]: I was a bit confused reading this at first. I don't think we need to repeat the supported shell script (we have it above) and we may not require a default as well

File: entrypoint.sh [hunk 3]
```diff
@@ -78,8 +90,9 @@ log_invalid_files(){
     for file in ${invalid_files[@]}; do
         printf "\n\t\e[33m %s \e[0m\n" "$file"
     done
-    printf "\n\e[33m ShellCheck only supports sh/bash/dash/ksh scripts. For supported scripts to be scanned, make sure to add a proper shebang on the first line of the script.\e[0m\n"
+    printf "\n\e[33m ShellCheck only supports sh/bash/dash/ksh scripts. For supported scripts to be scanned, make sure to add a proper shebang on the first line of the script.\n\n To fix the warning for the unsupported scripts, use the 'excluded_paths' input. For more information check:
```
Comment (asadmansr) [inline entrypoint.sh:93]: Regarding `To fix the warning for the unsupported scripts, use the 'excluded_paths' input`, should we also indicate that they can ignore files they don't want to lint (which could already a supported script). Something like `To ignore specific files or unsupported scripts, use the ...`?

File: tests/unit_tests/scan_tests.sh [hunk 4]
```diff
@@ -1,27 +1,30 @@
...
 }
 
 test_scan_valid_script_without_extension(){
     local expected="Scanning executable_script"
     local actual=$(scan_file ./test_data/test_dir/executable_script)
+
     assertContains "Actual messages:$actual Did not contain the expected message.\n" "$actual" "$expected" 
 }
 
 test_scan_unsuppoted_script(){
```
Comment (asadmansr) [inline tests/unit_tests/scan_tests.sh:21]: Oops, we missed a typo from previous PRs

Comment (Azbagheri) [reply tests/unit_tests/scan_tests.sh:21]: good catch!!

**Review by asadmansr — APPROVED** [review]

**Pull request merged**